pkg/kclient/all.go

```diff
@@ -49,13 +54,15 @@ func getAllResources(client dynamic.Interface, apis []apiResource, ns string, se
 				errResult = err
 				return
 			}
-			mu.Lock()
-			out = append(out, v...)
-			mu.Unlock()
+			outChan <- v
 			klog.V(4).Infof("[query api]  done: %s, found %d apis", a.GroupVersionResource(), len(v))
 		}(api)
 	}
 
+	for i := 0; i < len(apisOfInterest); i++ {
+		out = append(out, <-outChan...)
```

Comment:
This might be silly, but can you not nest this line to the for loop above that also iterates over apisOfInterest, or does it have something to do with go routines and channels?

Reply:
> This might be silly,

It's not silly. In fact, last time I made this exact change in the code (but didn't push it), it took me probably around an hour to get it right without introducing a deadlock.

And you are not wrong. It "just works" if I make the change you described, but...

> can you not nest this line to the for loop above that also iterates over apisOfInterest, or does it have something to do with go routines and channels?

But that would defeat the purpose of spinning up the go routines. If I move the line `out = append(out, <-outChan...)` just before the end of the previous `for` loop, it will be a blocking call that won't increment the `for` loop, and hence won't spin up the next go routine, till the current go routine has put something on `outChan`.

If you test against a local minikube/CRC instance, the performance issue won't be noticeable, but take a look at the below benchmark for a CRC VM running in North America to which I am connected via VPN:

```sh
# with code as-is in the PR
$ odo-pods-first logs | grep "===="
============================= 9.079955189s =============================
============================= 9.339786147s =============================

# after making the change you are wondering about
# basically remove the extra for loop and move the append statement to just after calling the go routine
$ odo logs | grep "===="
============================= 1m1.383878082s =============================
============================= 1m1.650332361s =============================

# I hear you say "make it a buffered channel instead of an unbuffered one"
# ok, here's the result after that change where I make outChan buffered
# by doing outChan := make(chan []unstructured.Unstructured, len(apisOfInterest))
# it's still a blocking call
$ odo logs | grep "===="
============================= 58.996780932s =============================
============================= 59.300285635s =============================
```

So an extra `for` loop is well worth it, IMO.